src/Microsoft.AspNet.Razor/CodeGenerators/CSharpCodeWriter.cs

```diff
@@ -445,18 +458,26 @@ private void WriteVerbatimStringLiteral(string literal)
         {
             Write("@\"");
 
-            foreach (char c in literal)
+            // We need to find the index of each '"' (double-quote) to escape it.
+            var start = 0;
+            int end;
+            while ((end = literal.IndexOf('\"', start)) > -1)
```

Comment:
With this new implementation, `WriteVerbatimStringLiteral()` should be much more efficient than `WriteCStyleStringLiteral()`, writing fewer / shorter `string`s when the `Writer` is a `StringWriter`. Given that, suggest following through on an earlier [PR #652 discussion](https://github.com/aspnet/Razor/pull/652#discussion-diff-48799035) i.e. remove the `Length` checks in the condition on [line 132](https://github.com/aspnet/Razor/pull/667/files#diff-d020ecaca6e34f77668b2ffe8de6d65aR132). That way ``WriteCStyleStringLiteral()` will only be called in the unusual condition that the literal includes a null `char`.